src/js/select2/dropdown/attachBody.js

```diff
@@ -32,6 +32,15 @@ define([
           self._resizeDropdown();
         });
       }
+
+	  function repositionOnSelectionChange() {
+	    if (!this.options.get('closeOnSelect')) {
```

Comment:
You may only want to reposition the dropdown if you can detect that the dropdown is still open. I think that may have been what you were going for here, but you can probably solve the issue by looking at the dropdown state instead of checking for a specific option.